Tests/swift-sdk-swift-tests/IterableActionRunnerTests.swift

```diff
@@ -45,29 +47,56 @@ class IterableActionRunnerTests: XCTestCase {
     
     func testUrlHandlingOverride() {
         let urlString = "https://example.com"
-        let urlDelegate = MockUrlDelegate(returnValue: true)
+        let action = IterableAction.action(fromDictionary: ["type" : "openUrl", "data" : urlString])!
+        let context = IterableActionContext(action: action, source: .push)
         let urlOpener = MockUrlOpener()
-        let actionRunner = IterableActionRunner(urlDelegate: urlDelegate, customActionDelegate: nil, urlOpener: urlOpener)
+
+        let handled = IterableActionRunner.execute(action: action,
+                                     context: context,
+                                     urlHandler: { url in return true},
+                                     urlOpener: urlOpener)
         
-        let action = IterableAction.action(fromDictionary: ["type" : "openUrl", "data" : urlString])!
-        _ = actionRunner.execute(action: action, from: .push)
+        XCTAssertTrue(handled)
         
         if #available(iOS 10.0, *) {
-            XCTAssertNil(urlOpener.preIos10openedUrl)
             XCTAssertNil(urlOpener.ios10OpenedUrl)
-        } else {
             XCTAssertNil(urlOpener.preIos10openedUrl)
+        } else {
             XCTAssertNil(urlOpener.ios10OpenedUrl)
+            XCTAssertNil(urlOpener.preIos10openedUrl)
         }
     }
     
     func testCustomAction() {
-        let customActionDelegate = MockCustomActionDelegate(returnValue: false)
-        let action = IterableAction.action(fromDictionary: ["type" : "customActionName"])!
-        let actionRunner = IterableActionRunner(urlDelegate: nil, customActionDelegate: customActionDelegate, urlOpener: MockUrlOpener())
-        _ = actionRunner.execute(action: action, from: .push)
+        let customActionName = "myCustomActionName"
+        let action = IterableAction.action(fromDictionary: ["type" : customActionName])!
+        let context = IterableActionContext(action: action, source: .push)
+        let customActionHandler: CustomActionHandler = {name in
+            XCTAssertEqual(name, customActionName)
```

Comment:
Can we verify that this handler was called? If `execute` just retuns `true` without calling `customActionHandler`, the test will still pass.

Reply:
Previously, we were using XCTest expectations for that. Not sure if there is a different API for Swift.